### PR TITLE
small fixes pointed by golint

### PR DESCRIPTION
**Pull request by owulveryck**
The purpose of this PR is to do small fixes pointed by the linter.
Most of them are about documentation.

see https://go-lint.appspot.com/gorgonia.org/gorgonia

### CODE REVIEW

**Comment by owulveryck** [comment]
Once the work is done, the only messages that should remain are those related to Public functions returning private objects.
But this goes beyond linting and the fixes should be postponed to a release (it will probably have an impact on the API)

File: utils.go [hunk 1]
```diff
@@ -139,10 +139,8 @@ func hasInf(v Value, dev Device) bool {
 	switch vt := v.(type) {
 	case *F64:
 		return false
```
Comment (chewxy) [inline utils.go]: This is incorrect

File: utils.go [hunk 2]
```diff
@@ -139,10 +139,8 @@ func hasInf(v Value, dev Device) bool {
...
-		return math.IsInf(float64(*vt), 0)
```
Comment (chewxy) [inline utils.go]: This is correct. 

I think I added the `return false` during some debug

File: batch.go [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
 	DoWork()
 }
 
+// BatchedBLAS interface describes any object that can process BLAS work in batch
 type BatchedBLAS interface {
 	Batched
 	BLAS
 }
 
+// BatchedDevice ...
```
Comment (owulveryck) [inline batch.go]: I don't know what this interface represents

File: complex_test.go [hunk 4]
```diff
@@ -25,16 +25,16 @@ func TestWeirdNetwork(t *testing.T) {
 	var p *Node
 	var q *Node
 
-	e_x := NewMatrix(g, Float64, WithShape(xs, embeddingDims), WithName("x embeddings"), WithInit(GlorotU(1)))
-	e_p := NewMatrix(g, Float64, WithShape(ps, embeddingDims), WithName("p embeddings"), WithInit(GlorotU(1)))
-	e_q := NewMatrix(g, Float64, WithShape(qs, embeddingDims), WithName("q embeddings"), WithInit(GlorotU(1)))
-	w0_x := NewMatrix(g, Float64, WithShape(hiddenSize, xFeats*embeddingDims), WithName("layer0 weights for x"), WithInit(GlorotU(1)))
-	w0_p := NewMatrix(g, Float64, WithShape(hiddenSize, pFeats*embeddingDims), WithName("layer0 weights for p"), WithInit(GlorotU(1)))
-	w0_q := NewMatrix(g, Float64, WithShape(hiddenSize, qFeats*embeddingDims), WithName("layer0 weights for q"), WithInit(GlorotU(1)))
+	eX := NewMatrix(g, Float64, WithShape(xs, embeddingDims), WithName("x embeddings"), WithInit(GlorotU(1)))
```
Comment (owulveryck) [inline complex_test.go]: I am aware that the underscore may have a meaning, but the linter doesn't agree

File: debug.go [hunk 5]
```diff
@@ -32,12 +33,9 @@ var (
 	allocatorDev      = false
 )
 
-var READMEMSTATS = true
-
+// TABCOUNT is a global flag used when debugging
```
Comment (owulveryck) [inline debug.go]: Don't know what it does

File: debug.go [hunk 6]
```diff
@@ -32,12 +33,9 @@ var (
 	allocatorDev      = false
 )
 
-var READMEMSTATS = true
```
Comment (owulveryck) [inline debug.go]: Did not find any reference anywhere, so I consider it as dead code

File: debug.go [hunk 7]
```diff
@@ -32,12 +33,9 @@ var (
...
-
+// TABCOUNT is a global flag used when debugging
 var TABCOUNT uint32
 
-var TRACK = false
```
Comment (owulveryck) [inline debug.go]: Did not find any reference anywhere, so I consider it as dead code

File: engine.go [hunk 8]
```diff
@@ -5,13 +5,12 @@ import (
 	"gorgonia.org/tensor"
 )
 
-type SymbolicEngine struct{}
```
Comment (owulveryck) [inline engine.go]: Did not find any reference anywhere, so I consider it as dead code.

File: noextern.go [hunk 9]
```diff
@@ -72,6 +72,7 @@ func (m *ExternMetadata) Transfer(toDev, fromDev Device, v Value, synchronous bo
 	return v, nil
 }
 
+// Reset is a noop function for compatibility with the interface
```
Comment (owulveryck) [inline noextern.go]: which interface btw?

File: op.go [hunk 10]
```diff
@@ -96,12 +96,6 @@ type BinaryOp interface {
 	IsBinary() bool
 }
 
-type BestDoer interface {
```
Comment (owulveryck) [inline op.go]: Already discussed: Did not find any reference anywhere, so I consider it as dead code

File: batch.go (hunk 3 above)
Comment (chewxy) [reply batch.go]: BatchedDevices are the superset of BatchedBlas and the batched CUDA workflow.

File: debug.go (hunk 5 above)
Comment (chewxy) [reply debug.go]: readmemstats is an old variable used to dump memory statistics. Not used any more

File: engine.go (hunk 8 above)
Comment (chewxy) [reply engine.go]: `SymbolicEngine` is for the upcoming v0.10.0 API

File: noextern.go (hunk 9 above)
Comment (chewxy) [reply noextern.go]: It's a fake "interface". Both `// +build cuda` and `// +build !cuda` requires a `Reset()` function,

**Comment by chewxy** [comment]
can I merge this?

**Comment by owulveryck** [comment]
Good to merge. Thank you.

**Pull request merged**